src/components/Underline/Underline.jsx

```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { partitionText } from '../../util/text-manipulation';
+import { lucidClassNames } from '../../util/style-helpers.js';
+
+const cx = lucidClassNames.bind('&-Underline');
+
+const { node, string, instanceOf, oneOfType } = PropTypes;
+
+const matchAllRegexp = /^.*$/;
+
+const Underline = ({ className, children, match }) => {
+	if (!_.isRegExp(match)) {
+		if (_.isString(match)) {
```

Comment:
this looks like a good candidate for a ternary
also, is it a good idea to mutate `match` instead of creating a local var?

Reply:
it doesn't mutate the actual match that is passed in, it just reassigns a new value to that reference. i could create a local var instead of reusing the reference if that makes it more clear.